packages/button/test/button.test.ts

```diff
@@ -513,6 +513,32 @@ describe('Button', () => {
 
         expect(el.active).to.be.false;
     });
+    it('shows pending state after delay', async () => {
+        const el = await fixture<Button>(
+            html`
+                <sp-button label="Button">
+                    <svg slot="icon"></svg>
+                </sp-button>
+            `
+        );
+
+        await elementUpdated(el);
+        el.pending = true;
+        await elementUpdated(el);
+
+        waitUntil(
+            () => {
+                el.disabled;
+            },
+            'button is disabled',
+            { interval: 200, timeout: 1200 }
+        );
+
+        await elementUpdated(el);
+
+        expect(el.disabled).to.be.true;
+        expect(el.pending).to.be.true;
```

Comment:
Using an accessibility snap shot here could support us in guaranteeing a specific entry in the a11y tree.

Reply:
Updated to use the a11y snapshot testing for the `aria-label` changing to the `pending-label` value.